docs/guides/deployment/dockerize-a-watt-app.md

```diff
@@ -0,0 +1,150 @@
+EXPOSE 3042
+
+# Start the application
+CMD npm run start
+```
+
+### Explanation
+- **WORKDIR /app**: Sets the working directory inside the container to /app, where all commands will be executed.
+- **COPY package.json .**: Copies the `package.json` file from the local directory to the `/app` directory in the container. It's important to do this for all files in each service. 
+- **RUN --mount=type=bind,source=./package.json,target=./package.json**: Installs dependencies for the main application using a [bind mount](https://docs.docker.com/engine/storage/bind-mounts/) for the `package.json` file.
+- **--mount=type=cache,target=/root/.npm**: Caches the node_modules in the specified directory to speed up subsequent builds.
+- **COPY . .**: Copies all remaining files and folders into the /app directory in the container.
+- **RUN npm run build**: Executes the build script defined in the `package.json`, which typically compiles assets and prepares the application for production.
+- **EXPOSE 3042**: Exposes port 3042, allowing external access to the application running in the container.
+- **CMD npm run start**: Specifies the command to start the application, using the start script defined in the `package.json`.
+
+
+## Dockerizing TypeScript Watt Application
+
+This guide will walk you through dockerizing a TypeScript Platformatic Watt Application.
+
+## Dockerfile for TypeScript Watt Application
+
+Ensure you have a `.dockerignore` file in your project root to avoid unnecessary files such as `node_modules`, `dist`, `.env`, and any other files that are not required being copied into your Docker image. Here is an example of a sample `.dockerignore` file:
+
+```sh 
+node_modules
+npm-debug.log
+Dockerfile
+.dockerignore
+.env
+*.log
+dist
+```
+
+This reduces the image size and speeds up the build process.
+
+## Dockerizing a TypeScript Watt Application
+
+For a TypeScript-based application, Dockerizing requires TypeScript compilation before deployment. Here’s how to set it up.
+
+### TypeScript Compilation
+
+Create a `tsconfig.json` to configure your TypeScript build process with the following settings: 
+
+```json
+{
+  "compilerOptions": {
+    "module": "commonjs",
+    "esModuleInterop": true,
+    "target": "es2020",
+    "sourceMap": true,
+    "pretty": true,
+    "noEmitOnError": true,
+    "incremental": true,
+    "strict": true,
+    "outDir": "dist",
+    "skipLibCheck": true
+  },
+  "watchOptions": {
+    "watchFile": "fixedPollingInterval",
+    "watchDirectory": "fixedPollingInterval",
+    "fallbackPolling": "dynamicPriority",
+    "synchronousWatchDirectory": true,
+    "excludeDirectories": [
+      "**/node_modules",
+      "dist"
+    ]
+  }
+}
+```
+
+Ensure `PLT_TYPESCRIPT=true` in your `.env` file for local development. For production, set `PLT_TYPESCRIPT=false` and compile TypeScript using:
+
+```sh
+npx platformatic compile
+```
+
+This step compiles your TypeScript files and outputs them to the specified `outDir`.
```

Comment:
how does it relate to docker? How can somebody do the compilation via Docker? It's worth mentioning it here.